src/system.rs

```diff
@@ -344,3 +775,172 @@ fn make_map(data: &str) -> Result<HashMap<&str, u64>> {
+        user, nice, system, idle, iowait, irq, softirq, steal, guest, guest_nice,
+    )
+}
+
+/// Returns information about cpu times usage.
+///
+/// `/proc/stat` contains the cpu times statistics
+pub fn cpu_times() -> Result<CpuTimes> {
+    let data = read_file(Path::new("/proc/stat"))?;
+    let lines: Vec<&str> = data.lines().collect();
+    let cpu_info = match info_cpu_line(lines[0]) {
+        Ok(cpu_info) => cpu_info,
+        Err(error) => return Err(error),
+    };
+    Ok(cpu_line_to_cpu_times(&cpu_info))
+}
+
+/// Returns information about cpu time usage on a Vec
+/// that contains information per cpu
+///
+/// '/proc/stat' contains the cpu times statistics
+pub fn cpu_times_percpu() -> Result<Vec<CpuTimes>> {
+    let data = read_file(Path::new("/proc/stat"))?;
+    let mut lines: Vec<&str> = data.lines().collect();
+    let mut cpu_times_vector: Vec<CpuTimes> = Vec::new();
+    // Remove the first line that contain the total cpu: "cpu"
+    lines.remove(0);
+    for line in lines {
+        if line.starts_with("cpu") {
+            let cpu_info = match info_cpu_line(line) {
+                Ok(cpu_info) => cpu_info,
+                Err(error) => return Err(error),
+            };
+            let cpu_time = cpu_line_to_cpu_times(&cpu_info);
+            cpu_times_vector.push(cpu_time);
+        }
+    }
+    Ok(cpu_times_vector)
+}
+
+/// Return a float representing the current system-wide
+/// CPU utilization as percentage.
+///
+/// Interval must be > 0 seconds.
+/// If interval < 0.1, the result of this function will be meaningless.
+/// The function compares system CPU times elapsed before and after
+/// the interval (blocking).
+///
+/// Use information contains in '/proc/stat'
+pub fn cpu_percent(interval: f64) -> Result<f64> {
+    Ok(cpu_times_percent(interval)?.busy_times())
+}
+
+/// Return a vector of floats representing the current system-wide
+/// CPU utilization as percentage.
+///
+/// Interval must be > 0.0 seconds.
+/// If interval < 0.1, the result of this function will be meaningless.
+/// The function compares system CPU times per CPU elapsed before and after
+/// the interval (blocking).
+///
+/// Use information contains in '/proc/stat'
+pub fn cpu_percent_percpu(interval: f64) -> Result<Vec<f64>> {
+    let cpu_percent_percpu = cpu_times_percent_percpu(interval)?;
+    let mut cpu_percents: Vec<f64> = Vec::new();
+
+    for cpu_percent in cpu_percent_percpu {
+        cpu_percents.push(cpu_percent.busy_times());
+    }
+    Ok(cpu_percents)
+}
+
+/// Return a CpuTimesPercent representing the current detailed
+/// CPU utilization as a percentage.
+///
+/// Interval must be > 0.0 seconds.
+/// If interval is < 0.1, the result of this function will be meaningless.
+/// The function compares all CPU times elapsed before and after
+/// the interval (blocking).
+///
+/// Use informations contains in '/proc/stat'
+pub fn cpu_times_percent(interval: f64) -> Result<CpuTimesPercent> {
+    if interval <= 0. {
```

Comment:
This block (line 908 to 918) gets used again in `cpu_times_percent_percpu` and should probably be moved into a private function.